src/monocle_apptrace/metamodel/spans/span_example.json

```diff
@@ -0,0 +1,120 @@
+{
+    "name": "llamaindex.retrieve",
+    "context": {
+        "trace_id": "0x93cd0bf865b3ffcc3cf9c075dc3e3797",
+        "span_id": "0x5d3f839e900bda24",
+        "trace_state": "[]"
+    },
+    "kind": "SpanKind.CLIENT",
+    "parent_id": "0x7a63d63e42ccac60",
+    "start_time": "2024-09-09T14:38:45.237182Z",
+    "end_time": "2024-09-09T14:38:45.620112Z",
+    "status": {
+        "status_code": "OK"
+    },
+    "attributes": {
+        "monocle.entity.count": 2,
+        "monocle.entity.1.name": "ChromaVectorStore",
+        "monocle.entity.1.type": "MonocleEntity.VectorDB.Chroma",
+        "monocle.entity.1.embedding-model-name": "BAAI/bge-small-en-v1.5",
+        "monocle.entity.2.name": "BAAI/bge-small-en-v1.5",
+        "monocle.entity.2.type": "MonocleEntity.Model.embedding",
+        "monocle.entity.2.model_name": "BAAI/bge-small-en-v1.5"
+    },
+    "events": [
+        {
+            "name": "ChromaVectorStore",
+            "timestamp": "timestamp",
+            "attributes": {
+                "data.context_input": "question: What is an americano?",
+                "data.context_output": "Coffee is a hot drink made from the roasted and ground seeds (coffee beans) of a tropical shrub\nA latte consists of one or more shots of espresso, served in a glass (or sometimes a cup), into which hot steamed milk is added\nAmericano is a type of coffee drink prepared by diluting an espresso shot with hot water at a 1:3 to 1:4 ratio, resulting in a drink that retains the complex flavors of espresso, but in a lighter way",
+                "metadata": {}
+            }
+        }
+    ],
+    "links": [],
+    "resource": {
+        "attributes": {
+            "service.name": "coffee-bot"
+        },
+        "schema_url": ""
+    }
+}
+{
+    "name": "llamaindex.openai",
+    "context": {
+        "trace_id": "0x93cd0bf865b3ffcc3cf9c075dc3e3797",
+        "span_id": "0x8b6363e1937a4d7b",
+        "trace_state": "[]"
+    },
+    "kind": "SpanKind.CLIENT",
+    "parent_id": "0x7a63d63e42ccac60",
+    "start_time": "2024-09-09T14:38:45.622174Z",
+    "end_time": "2024-09-09T14:38:46.514120Z",
+    "status": {
+        "status_code": "OK"
+    },
+    "attributes": {
+        "monocle.entity.count": 2,
+        "monocle.entity.1.name": "AzureOpenAI",
+        "monocle.entity.1.type": "MonocleEntity.Inference.Azure_OpenAI",
+        "monocle.entity.1.model_name": "gpt-35-turbo",
+        "monocle.entity.1.provider_name": "api.openai.com",
+        "monocle.entity.1.az_openai_deployment": "kshitiz-gpt",
+        "monocle.entity.1.inference_endpoint": "https://okahu-openai-dev.openai.azure.com/",
+
+        "monocle.entity.2.name": "gpt-35-turbo",
+        "monocle.entity.2.type": "MonocleEntity.Model.LLM",
+        "monocle.entity.2.model_name": "gpt-35-turbo",
+        "monocle.entity.2.temperature": 0.1
+    },
+    "events": [
```

Comment:
for the following code in Langchain, where the input attributes are provided by the app developer

```
least_to_most_chain = (
        {'context': itemgetter('question') | retriever,
        'q_a_pairs': itemgetter('q_a_pairs'),
        'question': itemgetter('question'),
        }
        | least_to_most_prompt
        | llm
        | StrOutputParser()
    )
  ```

the following events get generated
```
"events": [
                {
                    "name": "input",
                    "timestamp": "2024-07-17T05:00:32.441831Z",
                    "attributes": {
                        "question": "What is Task Decomposition?",
                        "q_a_pairs": "Question: - What is the definition of task decomposition?\n\nAnswer: Task decomposition is the process of breaking down a complex task into smaller, simpler steps or subtasks. "
                    }
                },
                {
                    "name": "output",
                    "timestamp": "2024-07-17T05:00:39.502836Z",
                    "attributes": {
                        "response": "Task Decomposition is the process of breaking down a complex task into smaller, simpler steps or subtasks. This technique is often used in project management and artificial intelligence systems to enhance understanding and performance on complex tasks."
                    }
                }
            ]
            
```